### DISCUSSION TITLE
How do I do a migration if my backend container won't start due to an error?

### DISCUSSION THREAD

**Discussion by tharrington** [fastapi/full-stack-fastapi-template#1345] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
$ docker compose exec backend bash
service "backend" is not running
```

### Description

I have an error that was caused by some model code. I have updated the models and I want to create a new migration, but the following command fails:

```
docker compose exec backend bash
service "backend" is not running
```

Looking at the logs, it's clear this is caused by my model not having a relationship done properly.

### Operating System

macOS

### Operating System Details

Mac M1

### Python Version

3.11.2

**Comment by eugsokolov** [comment]
see here on how to apply migrations

https://github.com/fastapi/full-stack-fastapi-template/issues/324#issuecomment-2338590601

**Comment by YuriiMotov** [comment, marked as answer]
In `docker-compose.override.yml` comment out the following lines
```
    command:
      - fastapi
      - run
      - --reload
      - "app/main.py"
```

and uncomment the line above
```
    # command: sleep infinity  # Infinite loop to keep container alive doing nothing
```

Then run `docker compose watch` and in new terminal, attach to the container with `docker compose exec backend bash`.
Now you can see something like
```
root@7f2607af31c3:/app#
```
You are inside the backend container and can run alembic commands

Docs: https://github.com/fastapi/full-stack-fastapi-template/blob/master/backend/README.md#docker-compose-override
